src/lib/components/eventModal.svelte

```diff
@@ -1,184 +1,241 @@
+                    ? selected.resource?.actions
+                    : selected.service?.actions;
+
+                // We need to check if the selected action name is still
+                // on the available list, and if so, change tje action to the
+                // appropriate one, as it may contain different attributes
+                const availableAction = availableActions.find(
+                    (a) => a.name === selected.action?.name
+                );
+
+                if (!availableAction) {
+                    selected.action = null;
+                    selected.attribute = null;
+                } else {
+                    selected.action = availableAction;
 
-        if (action) {
-            data.set('action', { value: action.name, description: 'action' });
+                    if (!selected?.action?.attributes?.includes(selected.attribute)) {
+                        selected.attribute = null;
+                    }
+                }
+                break;
+            }
+            case 'action': {
+                selected.attribute = null;
+                break;
+            }
         }
+    }
 
-        if (attribute && !resource) {
-            data.set('attribute', { value: attribute, description: 'attribute' });
-        } else if (action?.attributes && !resource) {
-            data.set('attribute', { value: '*', description: `attribute` });
+    function resetSelected() {
+        Object.keys(selected).forEach((key) => (selected[key] = null));
+    }
+
+    function inferSelectedFromCustomInput() {
+        resetSelected();
+        for (const field of customInput.split('.')) {
+            if (isService(field)) {
+                selected.service = services.find((s) => s.name === field);
+            } else if (isResource(field)) {
+                const resource = selected.service?.resources.find((r) => r.name === field);
+                selected.resource = resource;
+            } else if (isAction(field)) {
+                const action =
+                    selected.resource?.actions.find((a) => a.name === field) ||
+                    selected.service?.actions.find((a) => a.name === field);
+                selected.action = action;
+            } else if (isAttribute(field)) {
+                const attribute = selected.action?.attributes?.find((a) => a === field);
+                selected.attribute = attribute;
+            }
         }
-        return data;
     }
 
-    $: eventString = createEventString(
-        selectedService,
-        selectedResource,
-        selectedAction,
-        selectedAttribute
-    );
-    $: copyValue =
-        inputData ??
-        Array.from(eventString.values())
-            .map((d) => d.value)
-            .join('.');
-
-    $: if (selectedService) {
-        selectedResource = null;
-        selectedAction = null;
-        selectedAttribute = null;
+    function toggleShowInput(e?: Event) {
+        e?.preventDefault();
```

Comment:
if we don't always pass the event, it might make more sense to let svelte handle the `preventDefault` on `on:click` and remove the event here completely

Reply:
It's always passed now